controllers/toolchaincluster_resources/toolchaincluster_resources_controller.go

```diff
@@ -0,0 +1,68 @@
+package toolchaincluster_resources
+
+import (
+	"context"
+	"embed"
+	"fmt"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	applycl "github.com/codeready-toolchain/toolchain-common/pkg/client"
+	"github.com/codeready-toolchain/toolchain-common/pkg/template"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&toolchainv1alpha1.ToolchainCluster{}).
+		Complete(r)
```

Comment:
TBH I'm not 100% sure what this controller should watch for. Since it's suppose to be a generic controller that can apply potentially any kind of objects from manifests

Reply:
There are two options:
1. either don't create the whole `Reconciler` in toolchain-common, but let each caller (host/member) operator define what they want to watch. Toolchain-common could contain either some generic logic that would create the `Reconciler` based on the given values, or just the content of the `Reconcile` function
2. or parse the content of the template, go over all kinds that are in the template and add watch for each of the kind.

Reply:
Thanks for your suggestion, I'm implementing 2. in https://github.com/codeready-toolchain/toolchain-common/pull/373/commits/6c014d7e44734b2694b2c3f6fd0b3bbb695e7490